app/src/marketplace/components/ProductPage/MembersGraph/index.jsx

```diff
@@ -0,0 +1,145 @@
+        // calculate member count diff based on the type.
+        // E.g. 'join' with 3 addresses means +3 diff in member count.
+        let diffCoefficient = 0
+        if (data.type === 'join') {
+            diffCoefficient = 1
+        } else if (data.type === 'part') {
+            diffCoefficient = -1
+        } else {
+            // Reject other message types.
+            return
+        }
+
+        const entry = {
+            timestamp: metadata.messageId.timestamp,
+            diff: data.addresses.length * diffCoefficient,
+        }
+        setMemberData((oldArray) => [
+            ...oldArray,
+            entry,
+        ])
+    }, [])
+
+    useEffect(() => {
+        memberData.sort((a, b) => b.timestamp - a.timestamp)
+
+        // Only thing we know at the beginning is total member count
+        // at given timestamp.
+        const initialData = [{
+            x: memberCountUpdatedAt,
+            y: memberCount,
+        }]
+
+        // Because we cannot read the whole joinPartStream, we have to
+        // work backwards from the initial state and calculate graph points
+        // using the member count diff.
+        const data = memberData.reduce((acc, element, index) => {
+            acc.push({
+                x: element.timestamp,
+                y: acc[index].y + element.diff,
+            })
+            return acc
+        }, initialData)
+        setGraphData(data)
+    }, [memberData, memberCount, memberCountUpdatedAt])
+
+    useEffect(() => {
+        setMemberCountUpdatedAt(Date.now())
+    }, [memberCount])
+
+    useEffect(() => {
+        // Clear member data when we change shownDays because
+        // resubscription to stream will happen and data will
+        // be resent
+        setMemberData([])
+    }, [shownDays])
+
+    return (
+        <div className={className}>
+            <ClientProvider>
+                <Subscription
+                    key={`${joinPartStreamId}-${shownDays}`}
+                    uiChannel={{
+                        id: joinPartStreamId,
+                    }}
+                    resendFrom={Date.now() - (shownDays * MILLISECONDS_IN_DAY)}
+                    onMessage={onMessage}
+                    isActive
+                />
+            </ClientProvider>
+            <XYPlot
+                xType="time"
+                width={540}
+                height={200}
+                margin={{
+                    left: 0,
+                    right: 50,
+                }}
+            >
+                <HorizontalGridLines />
+                <LineSeries
+                    curve={null}
+                    color="#0324FF"
+                    opacity={1}
+                    strokeStyle="solid"
+                    strokeWidth="4"
+                    data={graphData}
+                />
+                <XAxis
+                    hideLine
+                    style={axisStyle}
+                    tickTotal={7}
```

Comment:
something like this might help format the x-axis:
https://github.com/streamr-dev/landing-page/blob/805aeba05328bf7165a03a2f1859b0b8420877a8/src/components/Top/Token/PriceGraph/index.jsx#L171